### PR TITLE
fix(no-wait-for-side-effects): false positive inside `.then()`

### PR DESCRIPTION
**Pull request by sjarva**
## Checks

- [x] I have read the [contributing guidelines](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/CONTRIBUTING.md).
- [ ] If some rule is added/updated/removed, I've regenerated the rules list (`npm run generate:rules-list`)
- [ ] If some rule meta info is changed, I've regenerated the plugin shared configs (`npm run generate:configs`)

## Changes

- when reporting side efdects, check if the node is a `.then()`, and if it is, not report it
- added unit tests
- adds a mention of edge case to rule documentation

## Context

Closes: #500

### CODE REVIEW

File: tests/lib/rules/no-wait-for-side-effects.test.ts [hunk 1]
```diff
@@ -722,6 +735,41 @@ ruleTester.run(RULE_NAME, rule, {
       `,
 				errors: [{ line: 4, column: 11, messageId: 'noSideEffectsWaitFor' }],
 			} as const,
+			{
+				// Issue #500, https://github.com/testing-library/eslint-plugin-testing-library/issues/500
+				code: `
+        import { waitFor } from '${testingFramework}';
+        await waitFor(function() {
```
Comment (Belco90) [inline tests/lib/rules/no-wait-for-side-effects.test.ts:742]: Could you remove the `await` operator from the `waitFor` utils chained with `then`? So we can make sure the example is valid because thenable promises are handled properly, and not because it was a side-effect from the `wait` operator.

File: lib/rules/no-wait-for-side-effects.ts [hunk 2]
```diff
@@ -56,6 +57,27 @@ export default createTestingLibraryRule<Options, MessageIds>({
...
+
+			const callExpressionNode = node.parent.parent as TSESTree.CallExpression;
+
+			const test =
+				isMemberExpression(callExpressionNode.callee) &&
+				ASTUtils.isIdentifier(callExpressionNode.callee.property) &&
+				callExpressionNode.callee.property.name === 'then';
+
+			return test;
+		}
```
Comment (Belco90) [inline lib/rules/no-wait-for-side-effects.ts:79]: I think you can reuse our [`hasChainedThen`](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/lib/node-utils/index.ts#L128) to simplify the chained then detection.

**Review by Belco90 — CHANGES_REQUESTED** [review]
Thanks for fixing this one too! Just a couple of small changes and should be ready to go.

File: tests/lib/rules/no-wait-for-side-effects.test.ts (hunk 1 above)
Comment (sjarva) [reply tests/lib/rules/no-wait-for-side-effects.test.ts:742]: Good catch! Done.

File: lib/rules/no-wait-for-side-effects.ts (hunk 2 above)
Comment (sjarva) [reply lib/rules/no-wait-for-side-effects.ts]: Good point this too. I couldn't get to work with `hasChainedThen`, but `hasThenProperty` did the job.

File: tests/lib/rules/no-wait-for-side-effects.test.ts (hunk 1 above)
Comment (sjarva) [reply tests/lib/rules/no-wait-for-side-effects.test.ts:742]: Removed the `await` also from the rule documentation example.

**Review by Belco90 — APPROVED** [review]
All good!
